loopy/kernel/__init__.py

```diff
@@ -1638,25 +1603,45 @@ def __ne__(self, other):
 
     # }}}
 
+    def get_copy_kwargs(self, **kwargs):
+        if "iname_to_tags" in kwargs:
+            assert "inames" not in kwargs
+            iname_to_tags = kwargs["iname_to_tags"]
+            domains = kwargs.get("domains", self.domains)
+            kwargs["inames"] = {name: Iname(name,
+                                            iname_to_tags.get(name, frozenset()))
+                                for name in _get_inames_from_domains(domains)
+                                }
+            del kwargs["iname_to_tags"]
+
+        if "domains" in kwargs:
+            inames = kwargs.get("inames", self.inames)
+            domains = kwargs["domains"]
+            kwargs["inames"] = {name: inames.get(name, Iname(name, frozenset()))
+                                for name in _get_inames_from_domains(domains)}
```

Comment:
This needs to do a better job merging the existing `self.inames` with new information:
- Existing `inames` might already have tags. It seems as though this blows those away?

Reply:
> This needs to do a better job merging the existing self.inames with new information:

But, doesn't the caller intend to "clear" the tags which they didn't pass. I.e. `LoopKernel.copy(iname_to_tags=iname_to_tags)` was always meant to be a drop-in replacement for all kernel's iname tags, right?